hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/utils/db/cache/CacheKey.java

```diff
@@ -53,4 +53,18 @@ public boolean equals(Object o) {
   public int hashCode() {
     return Objects.hash(key);
   }
+
+  @Override
+  public int compareTo(Object o) {
+    if(Objects.equals(key, ((CacheKey<?>)o).key)) {
+      return 0;
+    } else {
+      if (key instanceof String) {
+        return ((String) key).compareTo((String) ((CacheKey<?>)o).key);
+      } else {
+        // If not type string, convert to string and compare.
+        return key.toString().compareTo((((CacheKey<?>) o).key).toString());
```

Comment:
when can this happen?

Reply:
When CacheKey<KEY> KEY type is not string. For BucketTable type is string, for tokentable type is OzoneTokenIdentifier. As this is common class used by all tables in OM, added the if, else condition.

Reply:
I think the correct fix is to make OzoneTokenIdentifier a Comparable, and then enforce that the CacheKey implements Comparable.

Reply:
Makes sense. However let's see if there is an alternate way to do this. We should not have reflection for type-specific behavior. It is fragile.